### PR TITLE
(Refactor) Create condition for threshold, refactor cards to use it

### PR DESCRIPTION
**Pull request by theelk801**
I figured that threshold should have its own special condition especially with it coming back, I made this refactor a PR as a courtesy since it changes a lot of cards

### CODE REVIEW

File: Mage/src/main/java/mage/abilities/condition/common/ThresholdCondition.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        return Optional
+                .ofNullable(source)
+                .map(Controllable::getControllerId)
+                .map(game::getPlayer)
+                .map(Player::getGraveyard)
+                .map(HashSet::size)
+                .orElse(0) > 0;
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/condition/common/ThresholdCondition.java:26]: you might want this to be >=7

File: Mage/src/main/java/mage/abilities/condition/common/ThresholdCondition.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    }
+
+    @Override
+    public String toString() {
+        return "there are seven or more cards in your graveyard";
```
Comment (xenohedron) [inline Mage/src/main/java/mage/abilities/condition/common/ThresholdCondition.java:31]: for all threshold cards, the current oracle wording is "seven or more cards are in your graveyard"

(this wording is only used by Deep-Sea Terror and Illuminate History, which don't have the threshold ability word)

**Review by xenohedron — COMMENTED** [review]
I already have this open to check text discrepancies, so I'll finish it

File: Mage/src/main/java/mage/abilities/condition/common/ThresholdCondition.java (hunk 1 above)
Comment (theelk801) [reply Mage/src/main/java/mage/abilities/condition/common/ThresholdCondition.java:26]: yeah good call

File: Mage/src/main/java/mage/abilities/condition/common/ThresholdCondition.java (hunk 2 above)
Comment (theelk801) [reply Mage/src/main/java/mage/abilities/condition/common/ThresholdCondition.java:31]: I tried to copy what the previous value would do, I figure it's not worth worrying too much about the text discrepancies until after all the errata is applied anyway

Comment (xenohedron) [reply Mage/src/main/java/mage/abilities/condition/common/ThresholdCondition.java]: yeah no worries, I had just seen a couple dozen discrepancies on cards with threshold for a long time and never been motivated enough to fix them properly, and this is the perfect opportunity

**Pull request merged**